pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -474,7 +475,13 @@ public void registerListener(TopicName topicName, TopicPolicyListener<TopicPolic
 
     @Override
     public void unregisterListener(TopicName topicName, TopicPolicyListener<TopicPolicies> listener) {
-        listeners.computeIfAbsent(topicName, k -> Lists.newCopyOnWriteArrayList()).remove(listener);
```

Comment:
This change is not threadsafe.
You have to move your code inside the function passed to computeIfAbsent

Reply:
The function passed to computeIfAbsent will only be executed when topicName is NOT already in this `listeners`.
Consider the actually use case of the listener, we should not add a List value to the `listeners` map, right?
 
There are actually  two data structures that we should consider thread safe issue.
One is the ConcurrentHashMap `listeners`, two operations we used here,  `get` and `remove` are both threadSafe.
The other `topicListeners ` is the value of `listeners`, and its type is CopyOnWriteArrayList, with thread safe `remove` method.

One thing should be noticed is that it's `Collections.emptyList()` instead of `topicListeners` in line 482. This operation is atomic guaranteed by ConcurrentHashMap, So the key will be deleted if and only if the value list is empty. 

But it's all theoretical  analysis above, any cases that will lead to thread safe issue?

Reply:
It should be a problem if thread A reaches line 478 first, then thread B reaches line 473 and then add a listener. After that, thread A continues to remove the listener, finally, the added listener will be removed, but actually, the `unregisterListener` operation happens first, we will get unexpected behavior.

Reply:
Oh, I see. LGTM.